### PR TITLE
change dir/rot  and rot/dir to use algo

### PR DESCRIPTION
**Pull request by og76**
also:
 - rot to dir has axis to specify 
 - dir to rot outputs eul/quat/matrix if needed

the matrix is also directly 4x4 and normalized for matrix output
normalization is necessary for matrix out, or the scale will be affected too
not for eul, quat ...

(prev ver was not using this way, but 3x3 to 4x4 etc)

### CODE REVIEW

File: nodes/rotation/direction_to_rotation.py [hunk 1]
```diff
@@ -28,36 +31,18 @@ def draw(self, layout):
...
+        isLinked = self.getLinkedOutputsDict()
+        if not any(isLinked.values()): return ""
+        
+        rot = "eulerRotation" if isLinked["eulerRotation"] else ""
+        quat = "quaternionRotation" if isLinked["quaternionRotation"] else ""
+        mat = "matrixRotation" if isLinked["matrixRotation"] else ""
+
+        return generateDirectionToRotationCode("direction", "guide", 
+                                                self.trackAxis, self.guideAxis, 
+                                                matrixOutputName = "{}".format(mat),
```
Comment (JacquesLucke) [inline nodes/rotation/direction_to_rotation.py]: why not simply use: matrixOutputName = mat?

Comment (og76) [inline nodes/rotation/direction_to_rotation.py]: cause I need to normalize it for matrix, output, else the scale is affected too
only for that

Comment (og76) [inline nodes/rotation/direction_to_rotation.py]: for eul/quat normalized matrix is not a must, but for matrix, yes

Comment (JacquesLucke) [inline nodes/rotation/direction_to_rotation.py]: yes but where is the diference between:
"{}".format(mat)
mat

Comment (og76) [inline nodes/rotation/direction_to_rotation.py]: ah, ok
silly, got caught in the style ..

**Pull request merged**